src/librustdoc/html/render.rs

```diff
@@ -1503,13 +1529,17 @@ impl Context {
             self.recurse(name, |this| {
                 let item = item.take().unwrap();
 
-                let mut buf = Vec::new();
-                this.render_item(&mut buf, &item, false).unwrap();
-                // buf will be empty if the module is stripped and there is no redirect for it
-                if !buf.is_empty() {
-                    let joint_dst = this.dst.join("index.html");
-                    let mut dst = try_err!(File::create(&joint_dst), &joint_dst);
-                    try_err!(dst.write_all(&buf), &joint_dst);
```

Comment:
This section (and the related one above it) managed its own buffer since `render_item` would bypass calling `layout::render` or `layout::redirect` if the item was stripped and didn't need a page. Since i made a global buffer i needed to add some extra handling around this section to make sure it didn't copy the buffer to another buffer before actually writing it. Mainly to save the allocation and memcpy rather than the filesystem interactions.